test/tests/functional/pbs_trillion_jobid.py

```diff
@@ -444,3 +451,34 @@ def test_jobid_duplication(self):
         # Job should submit successfully because all existing id's has been
         # passed
         self.submit_job(lower=1, upper=2, job_id='3[]')
+
+    def test_jobid_resvid_after_multiple_restart(self):
+        """
+        Test to check the Jobid/Resvid should not wrap to 0 during
+        server restart multiple times consecutively either gracefully/abruptly
+        """
+        j = Job(TEST_USER)
+        jid = self.server.submit(j)
+        curr_id = int(jid.split('.')[0])
+        self.submit_job(job_id='%s' % str(curr_id + 1))
+        self.submit_job(lower=1, upper=2, job_id='%s[]' % str(curr_id + 2))
+        self.submit_resv(resv_id='R%s' % str(curr_id + 3))
+        # Gracefully stop and start the server twice consecutively
+        self.stop_and_restart_svr('normal')
+        self.stop_and_restart_svr('normal')
+        self.submit_job(job_id='%s' % str(curr_id + 4))
```

Comment:
you aren't really checking that the job id wrapped to 0 or not, wouldn't it be a simpler and more targeted test if you just submitted a single job after the server restarts and check that its id is not 0 ?

Reply:
self.submit_job() is a generic function and using throughout the whole testsuite in which we are checking the exact jobid numbers in other test cases.
So that's why used the same in this test also and if it will wrap to zero after server restart then testcase will fail anyway.
Checking the array jobs and reservation as well because they also gets the id from the same function.